website/docs/features/cors.mdx

```diff
@@ -0,0 +1,101 @@
+---
+id: cors
+title: Configuring CORS with Yoga
+sidebar_label: CORS
+---
+
+CORS stands for Cross Origin Resource Sharing. In a nutshell, as a security measure, browsers aren't allowed to access resources outside their own domain.
+
+If your api and web apps are deployed to different domains (or subdomains), you'll have to worry about CORS. For example, if your web client is deployed to example.com but your Yoga server is api.example.com. For security reasons your browser will not allow XHR requests (like the kind that the GraphQL client makes) to a domain other than the one currently in the browser's address bar.
+
+To deal with this you have two options:
+
+_1. Avoid CORS by proxying your requests_
+e.g. If you setup a proxy or redirect to forward requests from `example.com/api/*` to `api.example.com`, you avoid CORS issues all together.
+
+_2. Configure Yoga to send back CORS headers_
+Yoga comes with CORS support out of the box - CORS can be configured when creating the server either by passing a CORSOptions object, or a builder function that returns the CORSOptions object.
+
+```ts
+export type CORSOptions =
+  | {
+      origin?: string[] | string
+      methods?: string[]
+      allowedHeaders?: string[]
+      exposedHeaders?: string[]
+      credentials?: boolean
+      maxAge?: number
+    }
+  | false
+```
+
+### Example configuration using CORSOptions
+
+```
+import { createServer } from '@graphql-yoga/common'
+
+const yogaApp = createServer({
+  cors: {
+      origin: 'http://localhost:4000',
+      credentials: true,
+      allowedHeaders: ['X-Custom-Header'],
+      methods: ['POST'],
+  }
+  /* ...other args */
+})
+```
+
+This will return the following headers:
+
+```
+Access-Control-Allow-Origin: 'http://localhost:4000'
+Access-Control-Allow-Credentials: true
+Access-Control-Allow-Methods: POST
+Access-Control-Allow-Headers: X-Custom-Header
+```
+
+### Example configuration using builder function
+
+You can also pass a function to the cors property, that takes your request and contructs the options
+
+```ts
+import { createServer } from '@graphql-yoga/common'
+import { Request } from 'cross-undici-fetch'
+
+const yogaApp = createServer({
+  cors: (request: Request) => {
```

Comment:
I think we don't have to manually annotate the argument type of this function and can thus drop the import above as well, making this example less verbose.

Reply:
`Request` is globally available so this doesn't need an import already.